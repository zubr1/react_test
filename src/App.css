.App {
  text-align: center;
  padding-top: 200px;
}

.App .sign {
  padding: 20px;
}

/* I know it's dirty but this is just a fast solution */
.App .count {
  margin: 80px;
}