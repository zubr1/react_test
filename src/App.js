import React, { useState } from 'react';
import './App.css';

const initialState = {
  text: '',
  words: 0
}

function App() {
  const [leftTextArea, setLeftTextArea] = useState(initialState);
  const [rightTextArea, setRightTextArea] = useState(initialState);

  const countWords = (text) => {
    const trimmedText = text.trim();
    if(!trimmedText)
      return 0;

    return trimmedText.split(' ').length;
  }

  const onChangeLeftTextArea = (e) =>{
    const text = e.target.value;
    const words = countWords(text);
    setLeftTextArea({text, words})
  }  
  
  const onChangeRightTextArea = (e) =>{
    const text = e.target.value;
    const words = countWords(text);
    setRightTextArea({text, words})
  }

  const areEqual = leftTextArea.text === rightTextArea.text;

  return (
    <div className="App">
      <textarea
        value={leftTextArea.text}
        onChange={onChangeLeftTextArea}
      />
      {areEqual && <label className="sign">equal</label>}
      {!areEqual && <label className="sign">not equal</label>}
      <textarea
        value={rightTextArea.text}
        onChange={onChangeRightTextArea}
      />
      <div />
      <label className="count">Word count: {leftTextArea.words}</label>
      <label className="count">Word count: {rightTextArea.words}</label>
    </div>
  );
}

export default App;
