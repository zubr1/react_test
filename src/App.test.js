import App from './App';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// here should be more unit tests but I'm running out of time