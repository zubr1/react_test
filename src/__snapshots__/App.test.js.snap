// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <textarea
    onChange={[Function]}
    value=""
  />
  <label
    className="sign"
  >
    equal
  </label>
  <textarea
    onChange={[Function]}
    value=""
  />
  <div />
  <label
    className="count"
  >
    Word count: 
    0
  </label>
  <label
    className="count"
  >
    Word count: 
    0
  </label>
</div>
`;
